services/telemetry/base.go

```diff
@@ -106,3 +156,48 @@ func (s *Service) sendRequest(ctx context.Context, data []byte) error {
 	}
 	return nil
 }
+
+// GenerateUUID generates new UUID version 4 (random).
+func GenerateUUID() (string, error) {
+	uuid, err := uuid.NewRandom()
+	if err != nil {
+		return "", errors.Wrap(err, "can't generate UUID")
+	}
+
+	// Old telemetry IDs have only 32 chars in the table but UUIDs + "-" = 36
+	cleanUUID := strings.Replace(uuid.String(), "-", "", -1)
+	return cleanUUID, nil
+}
+
+// Currently we only detect OS (Linux distribution) version from the kernel version (/proc/version).
+// For both AMI and OVA this value is fixed by the environment variable and not autodetected.
+// If/when we decide to support installation with "normal" Linux package managers (apt, yum, etc.),
+// we could use the code that was there. See PM-1333 and PMM-1507 in both git logs and Jira for details.
+
+type pair struct {
+	re *regexp.Regexp
+	t  string
+}
+
+var procVersionRegexps = []pair{
```

Comment:
rules can become outdated or improved. in this case, we receive situation when old PMM version reports the wrong OS.
I believe it is better to pass full output as is to our server and analyze it centrally because in this case we always will have right analyze.
Even more, OS version is very important for RedHat/Debian distributions.
if we can analyze this string on our side, we can have kernel->OS version matching.

Reply:
> I believe it is better to pass full output as is to our server and analyze it
Sure, we will do it in the next version. Current server API is too limited for that.